File: calculator.py
def calculator(operation, a, b):
    """
    A basic calculator function that performs arithmetic operations.
    
    :param operation: String representing the operation ('add', 'subtract', 'multiply', 'divide')
    :param a: First number
    :param b: Second number
    :return: Result of the operation
    """
    if operation == 'add':
        return a + b
    elif operation == 'subtract':
        return a - b
    elif operation == 'multiply':
        return a * b
    elif operation == 'divide':
        if b != 0:
            return a / b
        else:
            return "Error: Division by zero!"
    else:
        return "Error: Invalid operation!"

File: calculator.ts
export class Calculator {
    add(a: number, b: number): number {
        return a + b;
    }

    subtract(a: number, b: number): number {
        return a - b;
    }

    multiply(a: number, b: number): number {
        return a * b;
    }

    divide(a: number, b: number): number {
        if (b === 0) {
            throw new Error("Quack! 🦆 Can't divide by zero!");
        }
        return a / b;
    }

    calculate(operation: string, a: number, b: number): number {
        switch (operation) {
            case 'add': return this.add(a, b);
            case 'subtract': return this.subtract(a, b);
            case 'multiply': return this.multiply(a, b);
            case 'divide': return this.divide(a, b);
            default:
                throw new Error("Quack-quack! 🦆 Unknown operation!");
        }
    }
}
